web/components/comments/comments-list.tsx

```diff
@@ -27,24 +28,40 @@ function contractPath(slug: string) {
 
 export function UserCommentsList(props: { user: User }) {
   const { user } = props
+  const pageSize = 50
+  const [pageNum, setPageNum] = useState(0)
+  const numComments = useNumUserComments(user.id)
+  const [pageComments, setPageComments] = useState<
+    {
+      key: {
+        contractId: any
+        contractQuestion: any
+        contractSlug: any
+      }
+      items: unknown[]
+    }[]
+  >([])
+  const [isLoading, setIsLoading] = useState(false)
 
-  const page = usePagination({ q: getUserCommentsQuery(user.id), pageSize: 50 })
-  const { isStart, isEnd, getNext, getPrev, getItems, isLoading } = page
-
-  const pageComments = groupConsecutive(getItems(), (c) => {
-    return {
-      contractId: c.contractId,
-      contractQuestion: c.contractQuestion,
-      contractSlug: c.contractSlug,
-    }
-  })
-
-  if (isLoading) {
-    return <LoadingIndicator className="mt-4" />
-  }
+  useEffect(() => {
```

Comment:
This works fine. However, it wouldn't keep any comments from previous pages in memory, so there will be latency going back and forth between pages.

One way to do that would be to keep a complete list of `items` and append to it for each new page loaded. Then, you would slice it based on the current page number.

If you figured this out, it would also be worth creating a hook for it so the code is reusable (like we had for firebase).

Reply:
will figure out later